src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationFixtureContentRootAttribute.cs

```diff
@@ -0,0 +1,79 @@
+    [AttributeUsage(AttributeTargets.Assembly, Inherited = false, AllowMultiple = true)]
+    public sealed class WebApplicationFixtureContentRootAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="WebApplicationFixtureContentRootAttribute"/>.
+        /// </summary>
+        /// <param name="key">
+        /// The key of this <see cref="WebApplicationFixtureContentRootAttribute"/>. This
+        /// key is used by <see cref="WebApplicationTestFixture{TStartup}"/> to determine what of the
+        /// <see cref="WebApplicationFixtureContentRootAttribute"/> instances on the test assembly should be used
+        /// to match a given Startup class.
+        /// </param>
+        /// <param name="contentRootPath">The path to the content root. This path can be either relative or absolute.
+        /// In case the path is relative, the path will be combined with
+        /// <code><see cref="Directory.GetCurrentDirectory()"/></code></param>
+        /// <param name="contentRootTest">
+        /// A file that will be use as a marker to determine that the content root path for the given context is correct.
+        /// </param>
+        /// <param name="priority">
+        /// The priority of this content root attribute compared to other attributes. When
+        /// multiple <see cref="WebApplicationFixtureContentRootAttribute"/> instances are applied for the
+        /// same key, they are processed with <see cref="int.Parse(string)"/>, ordered in ascending order and applied
+        /// in priority until a match is found.
+        /// </param>
+        public WebApplicationFixtureContentRootAttribute(
+            string key,
+            string contentRootPath,
+            string contentRootTest,
+            string priority)
+        {
+            Key = key;
+            ContentRootPath = contentRootPath;
+            ContentRootTest = contentRootTest;
+            Priority = priority;
+        }
+
+        /// <summary>
+        /// The key for the content root associated with this project. Typically <see cref="Assembly.FullName"/>.
+        /// </summary>
+        public string Key { get; }
+
+        /// <summary>
+        /// The content root path for a given project. This content root can be relative or absolute. If its a
+        /// relative path, it will be combined with <see cref="AppContext.BaseDirectory"/>.
+        /// </summary>
+        public string ContentRootPath { get; }
+
+        /// <summary>
+        /// A marker file used to ensure that the path the content root is being set to is correct.
+        /// </summary>
+        public string ContentRootTest { get; }
+
+        /// <summary>
+        /// A number for determining the probing order when multiple <see cref="WebApplicationFixtureContentRootAttribute"/>
+        /// instances with the same key are present on the test <see cref="Assembly"/>.
+        /// </summary>
+        public string Priority { get; }
```

Comment:
Maybe a silly question, but why isn't this an `int`?

Reply:
Cuz it's generated by MSBuild and it automatically quotes everything.

Reply:
Oh right it's from that task that adds assembly attributes. Got it.

Reply:
And I want to fail as late as possible, so that its easier to diagnose issues.

Reply:
Well it's easier to diagnose if it doesn't even compile! But for the other reason mentioned, that's not possible here.